farmdata2_modules/fd2_tabs/resources/NewCustomTableComponent.js

```diff
@@ -0,0 +1,352 @@
+ * @vue-event row-canceled - Emitted when a row was being edited and the cancel button is clicked.
+ * @vue-event {Number} row-deleted - Emitted when the delete button next to a row is clicked and the user confirms that they want to delete the row.  The payload indicates the 'id' of the row that is to be deleted.
+ */
+let NewCustomTableComponent = {
+    template:
+    `<span>
+        <button data-cy="export-btn" 
+            style="float: right; float: top;" class="btn fd2-red-button" @click="exportCSV" v-if="csvName != ''">Export
+        </button>
+        <div class="sticky-table">
+            <table data-cy="table" style="width:100%;" class="table table-bordered table-striped">
+                <thead>
+                    <tr class="sticky-header table-text" data-cy="table-headers">
+                        <th v-for="(column, hi) in columns"
+                        v-if="columns[hi].visible && column.inputType.type != 'boolean' && column.inputType.type != 'button'" 
+                        :data-cy="'h'+hi"
+                        style="text-align:center">{{ column.header }}</th>
+                        <th v-for="(column, hi) in columns"
+                        v-if="columns[hi].visible && column.inputType.type == 'boolean'"
+                        style="text-align:center"
+                        width=55
+                        :data-cy="'h'+hi"><input type="checkbox"></th>
+                        <th v-for="(column, hi) in columns"
+                        v-if="columns[hi].visible && column.inputType.type == 'button'"
+                        style="text-align:center"
+                        width=55
+                        :data-cy="'h'+hi"> {{ column.header }} </th>
+                        <th data-cy="edit-header" width=55 v-if="canEdit && !currentlyEditing">Edit</th>
+                        <th data-cy="save-header" width=55 v-if="canEdit && currentlyEditing">Save</th>
+                        <th data-cy="delete-header" width=55 v-if="canDelete && !currentlyEditing">Delete</th>
+                        <th data-cy="cancel-header" width=55 v-if="canDelete && currentlyEditing">Cancel</th>
+                    </tr>
+                </thead>
+                <tbody>
+                    <tr class="table-text" 
+                    v-for="(row, ri) in rows"
+                    :data-cy="'r'+ri">
+                        <td v-for="(item, ci) in row.data"
+                        v-if="columns[ci].visible"
+                        :data-cy="'td-r'+ri+'c'+ci"
+                        style="text-align:center">
+                            <div v-if="rowToEditIndex!=ri || columns[ci].inputType.type == 'no input'"
+                            :data-cy="'r'+ri+'c'+ci"

```

Comment:
This is the part that is rendering the table row if the row is not currently being edited, or if it does not have an input element type.  So when the row is not being edited, the check box element is not being rendered (same for the button.). This was fine when all elements were just text when they were not being edited.  But now that there are some that are not text (e.g. checkbox, button), you'll need to generalize this or add a special case to handle those elements for when a row is not being edited.

Reply:
I'm not sure I fully understand the issue here.  But from what you say, it seems that the buttons and check boxes are not disabled when editing a row unless the row is blank?  Is that accurate?   That doesn't seem like the behavior we want.  We'll want to sort this out if it is still not behaving correctly.  Please update on the current status.

Reply:
This shouldn't be the case anymore. This was when custom buttons were still part of the table row but has been moved to the top of the table. The radio buttons have been updated to be disabled at all times except for when that row is being edited. As far as I remember, rendering isn't an issue anymore.